src/amount.js

```diff
@@ -388,7 +388,7 @@ Amount.prototype._copy = function(value) {
 Amount.prototype.compareTo = function(to) {
   var toAmount = Amount.from_json(to);
   if (!this.is_comparable(toAmount)) {
-    return new Amount(NaN);
+    return 0;
```

Comment:
The amount class uses a mixture of NaN's and exceptions; there's a task to make it consistent: https://ripplelabs.atlassian.net/browse/RLJS-217

I think an exception would be much better than returning an incorrect result.